Gcpe.Hub.API/Controllers/PostsController.cs

```diff
@@ -69,18 +71,27 @@ public IActionResult GetAllPosts([FromQuery] NewsReleaseParams postParams)
             }
         }
 
+        const int checkInterval = 60;
         [HttpGet("Latest/{numDays}")]
         [Produces(typeof(IEnumerable<Models.Post>))]
+        [ProducesResponseType(304)]
         [ProducesResponseType(400)]
-        [ResponseCache(Duration = 300)] // change to 10 when using swagger
+        [ResponseCache(Duration = checkInterval)]
         public IActionResult GetLatestPosts(int numDays)
         {
             try
             {
-                IQueryable<NewsRelease> latest = QueryPosts().OrderByDescending(p => p.PublishDateTime);
-                latest = isProduction ? latest.Where(p => p.PublishDateTime >= DateTime.Today.AddDays(-numDays)) : latest.Take(20); // for testing with a stale db
+                IQueryable<NewsRelease> latest = isProduction ? QueryPosts().Where(p => p.PublishDateTime >= DateTime.Today.AddDays(-numDays))
+                                                              : QueryPosts().OrderByDescending(p => p.PublishDateTime).Take(20); // 20 for testing with a stale db
 
-                return Ok(latest.Select(p => p.ToModel(mapper)).ToList());
+                var now = DateTime.Now;
+                if (lastModifiedNextCheck <= now)
+                {
+                    lastModified = latest.OrderByDescending(p => p.Timestamp).FirstOrDefault()?.Timestamp.LocalDateTime;
+                    lastModifiedNextCheck = now.AddSeconds(checkInterval);
```

Comment:
how come this controller doesn't use the same principle as the others and also contains a 'lastModifiedNextCheck'?